app-crypt/veracrypt/veracrypt-1.19.ebuild

```diff
@@ -0,0 +1,118 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+inherit flag-o-matic linux-info pax-utils toolchain-funcs wxwidgets
+
+DESCRIPTION="Disk encryption with strong security based on TrueCrypt"
+HOMEPAGE="https://veracrypt.codeplex.com"
+SRC_URI="https://github.com/veracrypt/VeraCrypt/archive/VeraCrypt_${PV}.tar.gz"
+
+#### well, it's actually VeraCrypt license, which states "VeraCrypt is governed by the TrueCrypt License version 3.0"
+LICENSE="truecrypt-3.0"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="X +asm"
+RESTRICT="mirror bindist"
+
+WX_GTK_VER="3.0"
+
+RDEPEND="
+	>=sys-fs/lvm2-2.02.45
+	sys-fs/fuse
+	x11-libs/wxGTK:${WX_GTK_VER}[X?]
+	app-admin/sudo
+	dev-libs/pkcs11-helper
+"
+DEPEND="
+	${RDEPEND}
+	virtual/pkgconfig
+	asm? ( dev-lang/nasm )
+"
+
+S="${WORKDIR}/VeraCrypt-VeraCrypt_${PV}/src"
+
+pkg_setup() {
+	local CONFIG_CHECK="~BLK_DEV_DM ~DM_CRYPT ~FUSE_FS ~CRYPTO ~CRYPTO_XTS"
+	linux-info_pkg_setup
+
+	if use X; then
+		need-wxwidgets unicode
+	else
+		need-wxwidgets base-unicode
+	fi
+}
+
+PATCHES=(
+	# TODO: remove in >=1.20
+	"${FILESDIR}/${PV}/makefile-archdetect.diff"
+	"${FILESDIR}/${PV}/execstack-fix.diff"
+	"${FILESDIR}/${PV}/link-with-libdl.patch"
+	"${FILESDIR}/${PV}/remove-packaging-from-makefile.patch"
+)
+
+src_prepare() {
+	if use X; then
+		# Fix linking when NOGUI=1
+		sed -e "s/WX_CONFIG_LIBS := base/&,core/" -i Main/Main.make || die "sed Main/Main.make failed"
+	fi
+	default
+}
+src_compile() {
+	local EXTRA
+
+	use X || EXTRA+=" NOGUI=1"
+	use asm || EXTRA+=" NOASM=1"
+	append-flags -DCKR_NEW_PIN_MODE=0x000001B0 -DCKR_NEXT_OTP=0x000001B1
+# please use append-cppflags for preprocessor flags
+	emake \
+		${EXTRA} \
+		NOSTRIP=1 \
+		NOTEST=1 \
+		VERBOSE=1 \
+		CC="$(tc-getCC)" \
+		CXX="$(tc-getCXX)" \
+		AR="$(tc-getAR)" \
+		RANLIB="$(tc-getRANLIB)" \
+		TC_EXTRA_CFLAGS="${CFLAGS}" \
+		TC_EXTRA_CXXFLAGS="${CXXFLAGS}" \
+		TC_EXTRA_LFLAGS="${LDFLAGS}" \
+		WX_CONFIG="${WX_CONFIG}"
+}
+
+src_test() {
+	"${S}/Main/veracrypt" --text --test || die "tests failed"
+}
+
+src_install() {
+	dobin Main/veracrypt
+	dodoc Readme.txt "Release/Setup Files/VeraCrypt User Guide.pdf"
+	newinitd "${FILESDIR}/${PN}.init" ${PN}
+
+	if use X; then
+		newicon Resources/Icons/VeraCrypt-48x48.xpm veracrypt.xpm
+		make_desktop_entry ${PN} "VeraCrypt" ${PN} "System"
+	fi
+
+	pax-mark -m "${D}/usr/bin/veracrypt"
+}
+
+pkg_postinst() {
+	elog "There is an init script for VeraCrypt for Baselayout-2."
```

Comment:
Per https://www.gentoo.org/support/news-items/2012-02-14-baselayout-1-deprecation.html, baselayout-1 is long gone, we need to adjust this message accordingly.